### ISSUE TITLE
namespaced events and states can be confused

### ISSUE THREAD

**Issue by rosskevin** [state-machines/state_machines-audit_trail#1]
This is particularly confusing with multiple state_machines.  An event called by `#canceled_payment` in the `payment` namespace should not be possibly confused with `canceled` in another, though both would be logged as such in the database.

It appears that `#state_name` (with no namespace) and `#payment_state_name` (in a `payment` namespaced state_machine) both yield `canceled` instead of a full state name like you would expect from the generated event method name.

**Comment by rosskevin** [comment]
The best options appears to be to add the `namespace` field to the state transition.

**Comment by seuros** [comment]
:+1:

**Comment by seuros** [comment]
what about `machine` or `machine_name` ?

**Comment by rosskevin** [comment]
Well, since the attribute on machine is `namespace`, I figured it might be best to label it as such.

``` ruby
 state_machine :alarm_state, initial: :active, namespace: :'alarm' do
```

So our field/value is `namespace: 'alarm'`

**Comment by seuros** [comment]
:shipit:

**Comment by rosskevin** [comment]
resolved with commit 61548c0bfb4ed49291f06d412b264b8a8c188708 and released as 1.0.1
